### PR TITLE
[EZ] Update mypy to 1.9.0

### PR DESCRIPTION
**Pull request by malfet**
TODO:
 - Add linter that keeps `requirements-ci.txt` and `.lintrunner.toml` in sync

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: .ci/docker/requirements-ci.txt [hunk 1]
```diff
@@ -85,10 +85,10 @@ librosa>=0.6.2 ; python_version < "3.11"
 #Pinned versions:
 #test that import:
 
-mypy==1.8.0
+mypy==1.9.0
 # Pin MyPy version because new errors are likely to appear with each release
 #Description: linter
-#Pinned versions: 1.8.0
+#Pinned versions: 1.9.0
```
Comment (Skylion007) [inline .ci/docker/requirements-ci.txt:91]: You need to update the lintrunner.toml version too.

Comment (malfet) [reply .ci/docker/requirements-ci.txt:91]: Then it's even more mysterious why it's causing any problems...

**Comment by malfet** [comment]
List of changed package versions:
```
--- freeze-1.8.0.txt	2024-04-10 18:22:18
+++ freeze-1.9.0.txt	2024-04-10 18:21:57
@@ -11,7 +11,7 @@
 click==8.1.7
 colorama==0.4.6
 dill==0.3.7
-execnet==2.0.2
+execnet==2.1.1
 expecttest==0.1.6
 filelock==3.13.1
 flatbuffers==2.0
@@ -29,17 +29,17 @@
 iniconfig==2.0.0
 Jinja2==3.1.3
 jmespath==0.10.0
-joblib==1.3.2
+joblib==1.4.0
 junitparser==2.1.1
 lark==0.12.0
-lazy_loader==0.3
+lazy_loader==0.4
 lintrunner==0.10.7
 lxml==5.0.0
 Markdown==3.6
 MarkupSafe==2.1.5
 mpmath==1.2.1
 multidict==6.0.5
-mypy==1.8.0
+mypy==1.9.0
 mypy-extensions==1.0.0
 networkx==2.8.8
 ninja==1.10.2
@@ -70,7 +70,7 @@
 s3transfer==0.5.2
 scikit-image==0.20.0
 scikit-learn==1.4.1.post1
-scipy==1.12.0
+scipy==1.13.0
 setuptools==68.2.2
 six @ file:///tmp/build/80754af9/six_1644875935023/work
 sortedcontainers==2.4.0
@@ -81,9 +81,9 @@
 threadpoolctl==3.4.0
 tifffile==2024.2.12
 tlparse==0.3.7
-torch @ file:///var/lib/jenkins/workspace/dist/torch-2.4.0a0%2Bgit10d06fc-cp312-cp312-linux_x86_64.whl#sha256=b11e23acafbef502235bde60381e48f25f87f8546c6aec78e21021f152cbd30d
+torch @ file:///var/lib/jenkins/workspace/dist/torch-2.4.0a0%2Bgitf61b04a-cp312-cp312-linux_x86_64.whl#sha256=3088730024043e735220dd6c6de87280175d10955859286c314dc81fc16613a3
 torchvision @ git+https://github.com/pytorch/vision.git@2c4665ffbb64f03f5d18016d3398af4ac4da5f03
-typing_extensions==4.10.0
+typing_extensions==4.11.0
 unittest-xml-reporting==3.2.0
 urllib3==1.26.18
 Werkzeug==3.0.2
```

**Comment by malfet** [comment]
Oh, why the heck we are not pinning scipy

File: .ci/docker/requirements-ci.txt (hunk 1 above)
Comment (malfet) [reply .ci/docker/requirements-ci.txt:91]: Actually, we used to have a plugin that prints a warning about mypy discepancy as well as test, but looks like we've forgot about it. Which also makes me wonder: why the heck we still have mypy in `requirements-ci.txt`

**Pull request closed without merging**